README.md

```diff
@@ -123,10 +132,106 @@ To install pgai from source on a PostgreSQL server:
+    destination => <embedding_table_name>,
+    embedding => ai.embedding_openai(<model_name>, <dimensions>),
+    chunking => ai.chunking_recursive_character_text_splitter(<column_name>)
+);
+```
+This newly created vectorizer will automatically track any changes to the
+data in the source table and update the destination embedding table
+with the new embeddings asynchronously.
+
+[Automate AI embedding with pgai Vectorizer](/docs/vectorizer.md) shows you how
+to implement embeddings in your own data. When you install pgai on Timescale
+Cloud, embedding are automatically created and synchronized in the background.
+On a self-hosted Postgres installation, you use a [Vectorizer
+Worker](/docs/vectorizer-worker.md) to asynchronously processes your
+vectorizers.
+
+### Search your data using vector and semantic search
+
+pgai exposes a set of functions to directly interact with the LLM models through SQL, enabling 
+you to do semantic search directly in your database:
+
+```sql
+SELECT 
+   chunk,
+   embedding <=> ai.openai_embed(<embedding_model>, 'some-query') as distance
+FROM <embedding_table>
+ORDER BY distance
+LIMIT 5;
+```
+
+This is a perfectly normal SQL query. You can combine it with `where` clauses and other SQL features to 
+further refine your search. pgai solves the *missing where clause in vector search* problem for real.
+
+### Implement Retrieval Augmented Generation inside a single SQL statement
+
+Similar to [semantic search](#search-your-data-using-vector-and-semantic-search), pgai LLM functions 
+enable you to implement RAG directly in your database. For example:
+
+1. Create a RAG function:
+    ```sql
+    CREATE OR REPLACE FUNCTION generate_rag_response(query_text TEXT)
+    RETURNS TEXT AS $$
+    DECLARE
+       context_chunks TEXT;
+       response TEXT;
+    BEGIN
+       -- Perform similarity search to find relevant blog posts
+       SELECT string_agg(title || ': ' || chunk, ' ') INTO context_chunks
+       FROM (
+           SELECT title, chunk
+           FROM blogs_embedding
+           ORDER BY embedding <=> ai.openai_embed('text-embedding-3-small', query_text)
+           LIMIT 3
+       ) AS relevant_posts;
+
+       -- Generate a summary using gpt-4o-mini
+       SELECT ai.openai_chat_complete(
+           'gpt-4o-mini',
+           jsonb_build_array(
+               jsonb_build_object('role', 'system', 'content', 'You are a helpful assistant. Use only the context provided to answer the question. Also mention the titles of the blog posts you use to answer the question.'),
+               jsonb_build_object('role', 'user', 'content', format('Context: %s\n\nUser Question: %s\n\nAssistant:', context_chunks, query_text))
+           )
+       )->'choices'->0->'message'->>'content' INTO response;
+
+       RETURN response;
+    END;
+    $$ LANGUAGE plpgsql;
+    ```
+
+1. Execute your function in a SQL query:
+
+    ```sql
+    SELECT generate_rag_response('Give me some startup advice');-*_
+    ```
```

Comment:
Same here, should this get a result table?

Reply:
Good idea. Can I leave this to you in a followup pr?